app/models/proof_attempt.rb

```diff
@@ -1,6 +1,13 @@
 class ProofAttempt < ActiveRecord::Base
   belongs_to :theorem, foreign_key: 'sentence_id'
   belongs_to :proof_status
+  has_and_belongs_to_many :used_axioms,
+                          class_name: 'Sentence',
+                          join_table: 'used_axioms_proof_attempts'
+  has_and_belongs_to_many :used_theorems,
+                          class_name: 'Theorem',
+                          association_foreign_key: 'sentence_id',
+                          join_table: 'used_axioms_proof_attempts'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.